driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -21,116 +21,202 @@
 import java.util.function.Function;
 
 /**
```

Comment:
I think:

> A value that may be stored in or otherwise operated upon by MongoDB.

Followed by some less-dense variation on:

> This API is a language binding: it is a way to use a remote system as if
> it were a native Java API. It is a way to represent values and computations
> in some "execution context". This API is also a Domain Specific Language: it
> is the Java-native variant of MQL, which is used to query MongoDB or (more
> broadly) to work with data on MongoDB. 

It would be followed by a few paragraphs covering important points and gotchas.

Reply:
I added a long description (similar to [Stream](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html)). Unsure which parts of it, if any, might be moved to the package docs (none are obviously unrelated to the type hierarchy of which this is the root).